### PR TITLE
Addressing comments from bast on the repository

### PR DESCRIPTION
**Pull request by zachcran**
**Is this pull request associated with an issue(s)?**
Please list issues associated with this pull request, including [closing words](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword)
as appropriate.

https://github.com/openjournals/joss-reviews/issues/5711

**Description**

Reviewer @bast has raised some concerns and comments on CMakePPLang and the paper. This PR represents changes made in the code base and documentation, merging into the `master` branch. #114 addresses changes made in the paper, merging into https://github.com/CMakePP/CMakePPLang/pull/87 and not intended to merge into `master`.

1. The strategy for managing versioning, compatibility, deprecation, and API changes in this tool is not clear in the paper or code documentation.
2. It is not clear what CTOR means in Figures 2 and 3 in the paper.
3. In the paper, a short example showing the advantage of CMakePPLang's strong typing over CMake's weak typing would be useful.
4. The CMakePPLang version listed on the documentation website, https://cmakepp.github.io/CMakePPLang/, shows `1.0.0`, which is not the current version of CMakePPLang that is released.
5. The [install instructions](https://github.com/CMakePP/CMakePPLang?tab=readme-ov-file#installation-instructions) have an incorrect link to the CMakePPLang repository.

This PR partially addresses 1 by:
- including instructions about how to pin a specific CMakePPLang version,
- adding discussion to the README and documentation about the logic behind the current versioning scheme and how it will be incremented going forward,
- adding instructions to the README and documentation on how to pin a specific CMakePPLang release version

Although 2 refers to the paper, this PR also partially addresses it by clarifying what CTOR means in the "Example Usage" section of `README.rst`. Comment 2 is primarily addressed in #114.

Comment 3 is addressed in #114.

To address comment 4, the version on https://cmakepp.github.io/CMakePPLang/ is a target version for release `1.0.0`. It will need to start being updated with each release after the 1.0 release, which we plan to do with the Python module for git, using it to set the version off of the git tags in the `docs/source/conf.py` file.

This PR addresses 5 by fixing the incorrect link, https://github.com/CMakePP/cmakepp_lang, to the correct repository link, https://github.com/CMakePP/CMakePPLang. This seems to be a typo since `cmakepp_lang` was being used (correctly) everywhere around it.

**TODOs**

- [x] Respond to 1.
  - [x] ~Consider scheduling a weekly build/test action with a badge in the README to indicate if the latest version is stable.~ Moved to issue #116.
- [x] Respond to 4.

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -43,14 +43,30 @@ CMakePPLang can be included in a CMake project by adding:
...
-      GIT_REPOSITORY https://github.com/CMakePP/cmakepp_lang
+      GIT_REPOSITORY https://github.com/CMakePP/CMakePPLang
    )
    FetchContent_MakeAvailable(cmakepp_lang)
 
 to your ``CMakeLists.txt`` file. CMakePPLang will then be downloaded as part
 of the CMake configuration step. More detailed instructions, can be found
 `here <https://cmakepp.github.io/CMakePPLang/getting_started/obtaining_cmakepplang.html>`__.
 
+Information about pinning a specific version of CMakePPLang can be found
```
Comment (zachcran) [inline README.rst:54]: @ryanmrichard Should we point to the documentation for version pinning or include the short blurb commented out below this to include the instructions explicitly in the README?

**Review by ryanmrichard — APPROVED** [review]
I think what you have is perfect!

**Comment by ryanmrichard** [comment]
> Consider scheduling a weekly build/test action with a badge in the README to indicate if the latest version is stable.

I would just make this an issue for now unless you have a really quick fix. I want to get the 1.0 release out as soon as possible, we can add new features after that.

**Comment by zachcran** [comment]
> > Consider scheduling a weekly build/test action with a badge in the README to indicate if the latest version is stable.
> 
> I would just make this an issue for now unless you have a really quick fix. I want to get the 1.0 release out as soon as possible, we can add new features after that.

This has been moved to issue #116 for now.

**Comment by bast** [comment]
Thanks for these changes!

**Pull request merged**